source/jormungandr/jormungandr/scenarios/new_default.py

```diff
@@ -1243,6 +1243,19 @@ def add_olympics_forbidden_uris(origin_detail, destination_detail, api_request,
         api_request["forbidden_uris[]"] = instance.olympics_forbidden_uris.pt_object_olympics_forbidden_uris
 
 
+def add_additional_parameters(origin_uri, destination_uri, api_request, instance):
+    # Add parameter allowed_id[] in the request if instance.od_allowed_ids exists and allowed_ids found for a key
+    if not instance.od_allowed_ids:
+        return
+    allowed_ids = instance.get_od_allowed_ids(origin_uri, destination_uri)
+    if not allowed_ids:
+        return
+    if api_request.get("allowed_id[]"):
+        api_request["allowed_id[]"] += allowed_ids
```

Comment:
I think here you should also do
```suggestion
        api_request["allowed_id[]"] = allowed_ids
```
and completely ignore the allowed_id[] provided by the user.
Otherwise, you may obtain journeys that uses lines/routes that are not in the `od_allowed_ids`

Reply:
You are right. We are discussing about this point.
Thanks

Reply:
we re-initialize allowed_id[] values ignoring already existing parameter values